pages/app/post/[id]/settings.tsx

```diff
@@ -5,18 +5,41 @@ import CloudinaryUploadWidget from "@/components/Cloudinary";
 import LoadingDots from "@/components/app/loading-dots";
 import saveImage from "@/lib/save-image";
 import Modal from "@/components/Modal";
-import { useState, useEffect } from "react";
+import { useState, useEffect, FormEvent } from "react";
 import { useRouter } from "next/router";
 import Loader from "@/components/app/Loader";
 import toast, { Toaster } from "react-hot-toast";
 import { fetcher } from "@/lib/fetcher"
+import { Post, Site } from '@prisma/client';
+
+interface PostSettings {
+  slug: string;
+  id: string;
+  image: string;
+  imageBlurhash: string;
+}
+
+interface PostWithSite {
+  id: string;
+  title: string | null;
+  description: string | null;
+  content: string | null;
+  slug: string;
+  image: string | null;
+  imageBlurhash: string | null;
+  createdAt: Date;
+  updatedAt: Date;
+  published: boolean;
+  siteId: string | null;
+  site: Site | null;
+}
```

Comment:
This could probably be instead done by extending the generated Prisma types

```tsx
import type { Post, Site } from ".prisma/client";
```

```suggestion
interface PostWithSite extends Post {
  site: Site | null;
}
```

Reply:
Great call! Addressed in dd741ad9246faf78e13d133cf0905985511ea070